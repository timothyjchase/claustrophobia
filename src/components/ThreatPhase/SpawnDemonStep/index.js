import React from 'react'
import PropTypes from 'prop-types'
import renderHTML from 'react-render-html'
import { Button, Message } from 'semantic-ui-react'

const SpawnDemonStep = ({
  scenario,
  demonName,
  legalPlacement,
  completeThreatSpawnDemonStep,
}) => {
  const spawnLocation =
    scenario.demonSpawnLocation ||
    '<li>the <strong>largest group</strong> of Human warriors</li>'
  return (
    <div>
      {!legalPlacement && (
        <Message negative>
          <Message.Header>A Taste for Blood</Message.Header>
          <Message.Content>
            Human warriors cannot prevent the appearance of Demons.
          </Message.Content>
        </Message>
      )}
      <p>
        Add the
        <strong>{` ${demonName} `}</strong>
        to the tile closest to:
      </p>
      <ul>{renderHTML(spawnLocation)}</ul>
      {legalPlacement && (
        <p>
          <i>Reminder: Human warriors block tiles with unexplored openings</i>
        </p>
      )}
      <Button.Group vertical fluid>
        <Button
          primary
          icon="play"
          content="Next Phase"
          onClick={completeThreatSpawnDemonStep}
        />
      </Button.Group>
    </div>
  )
}

SpawnDemonStep.defaultProps = {
  demonName: 'Demon',
}

SpawnDemonStep.propTypes = {
  scenario: PropTypes.object.isRequired,
  demonName: PropTypes.string,
  legalPlacement: PropTypes.bool.isRequired,
  completeThreatSpawnDemonStep: PropTypes.func.isRequired,
}

export default SpawnDemonStep
