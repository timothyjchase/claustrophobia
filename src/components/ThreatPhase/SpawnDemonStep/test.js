import React from 'react'
import SpawnDemonStep from './index'

describe('<SpawnDemonStep />', () => {
  describe('with location, legal placement', () => {
    const props = {
      scenario: { demonSpawnLocation: 'custom location' },
      legalPlacement: true,
      completeThreatSpawnDemonStep: jest.fn(),
    }
    const component = shallow(<SpawnDemonStep {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with no location, no legal placement', () => {
    const props = {
      scenario: {},
      legalPlacement: false,
      completeThreatSpawnDemonStep: jest.fn(),
    }
    const component = shallow(<SpawnDemonStep {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
