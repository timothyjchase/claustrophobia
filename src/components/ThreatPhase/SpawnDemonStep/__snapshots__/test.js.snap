// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpawnDemonStep /> with location, legal placement should match snapshot 1`] = `
<div>
  <p>
    Add the
    <strong>
       Demon 
    </strong>
    to the tile closest to:
  </p>
  <ul>
    custom location
  </ul>
  <p>
    <i>
      Reminder: Human warriors block tiles with unexplored openings
    </i>
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<SpawnDemonStep /> with no location, no legal placement should match snapshot 1`] = `
<div>
  <Message
    negative={true}
  >
    <MessageHeader>
      A Taste for Blood
    </MessageHeader>
    <MessageContent>
      Human warriors cannot prevent the appearance of Demons.
    </MessageContent>
  </Message>
  <p>
    Add the
    <strong>
       Demon 
    </strong>
    to the tile closest to:
  </p>
  <ul>
    <li
      key="0"
    >
      the 
      <strong
        key="1"
      >
        largest group
      </strong>
       of Human warriors
    </li>
  </ul>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;
